test/config/config.test.js

```diff
@@ -504,6 +533,69 @@ const testFixtures = [
       );
     },
   },
+  {
+    name: 'use agent - should support string, regex & wildcards in ignore options',
+    script: 'fixtures/use-agent.js',
+    cwd: __dirname,
+    noConvenienceConfig: true,
+    env: {
+      ELASTIC_APM_TRANSACTION_IGNORE_URLS: 'foo,bar,/wil*card',
+      ELASTIC_APM_ELASTICSEARCH_CAPTURE_BODY_URLS: '*/_search,*/_eql/search',
+      TEST_APM_START_OPTIONS: JSON.stringify(
+        {
+          ignoreUrls: ['str1', /regex1/],
+          ignoreUserAgents: ['str2', /regex2/],
+        },
+        replacer,
+      ),
+    },
+    checkScriptResult: (t, err, stdout) => {
+      t.error(err, `use-agent.js script succeeded: err=${err}`);
+      const useAgentLogs = getUseAgentLogs(stdout);
+      const resolvedConfig = JSON.parse(useAgentLogs[2], reviver);
+
+      t.deepEqual(
+        resolvedConfig.transactionIgnoreUrls,
+        ['foo', 'bar', '/wil*card'],
+        'transactionIgnoreUrls is parsed correctly from environment (wildcards)',
+      );
+      t.deepEqual(
+        resolvedConfig.transactionIgnoreUrlRegExp,
+        [/^foo$/i, /^bar$/i, /^\/wil.*card$/i],
+        'transactionIgnoreUrlRegExp is parsed correctly from environment (wildcards)',
+      );
+      t.deepEqual(
+        resolvedConfig.elasticsearchCaptureBodyUrls,
+        ['*/_search', '*/_eql/search'],
+        'elasticsearchCaptureBodyUrls is parsed correctly from environment (wildcards)',
+      );
+      t.deepEqual(
+        resolvedConfig.elasticsearchCaptureBodyUrlsRegExp,
+        [/^.*\/_search$/i, /^.*\/_eql\/search$/i],
+        'elasticsearchCaptureBodyUrls is parsed correctly from environment (wildcards)',
+      );
+      t.deepEqual(
+        resolvedConfig.ignoreUrlStr,
+        ['str1'],
+        'string items of ignoreUrl are added to the right config (ignoreUrlStr)',
```

Comment:
```suggestion
        'string items of ignoreUrls are added to the right config (ignoreUrlStr)',
```
nit. Also a case below.